### ISSUE TITLE
Launching apps from split tunneling does not change DNS. This leads to websites like Netflix to show a 403 error even though the browser is not using the vpn connection

### ISSUE THREAD

**Issue by adityasubathu** [mullvad/mullvadvpn-app#2284]
## Issue report

Operating system: Manjaro 20.2

App version: 2020.7

### Issue description

Launching apps from split tunneling does not change DNS. This leads to websites like Netflix to show a 403 error even though the browser is not using the vpn connection

**Comment by faern** [comment]
Yes we are aware of this behaviour. It's currently by design. Because DNS is messy and DNS management on Linux is an even messier story. But we are working on improving the feature iteratively. Maybe split tunneling on Linux will be able to split the DNS in the future. It also depends a bit on how your system manages DNS. NetworkManager, systemd-resolved, static `/etc/resolv.conf` etc etc. There are many variables. For now we decided to make it consistenly do DNS inside the tunnel to at least be easy to reason about it.

**Comment by adityasubathu** [comment]
I have NextDNS CLI installed and set up on my machine. It used to work with mullvad enabled until 2020.7. Downgrading to 2020.6 did not solve this issue. Is there a way I can use NextDNS with Mullvad?

**Comment by faern** [comment]
Nope. Not yet. The current versions of the app, when connected, will force the use of the VPN relay server as the one and only DNS resolver. Not sure how you got it working with older releases.

But in the master branch we have implemented custom DNS. Then it will be possible to configure it to use another DNS resolver when connected. But both split and non split apps will use that same resolver.

**Comment by adityasubathu** [comment]
> But in the master branch we have implemented custom DNS

So in the next release we will be able to set custom DNS?

> But both split and non split apps will use that same resolver

Yeah that won't be an issue.

**Comment by faern** [comment]
> So in the next release we will be able to set custom DNS?

Exactly. Unless there is some problem during testing that forces us to back that feature out. But the plan is to launch custom DNS with most platforms in the coming release.
